frontend/public/components/dashboards-page/overview-dashboard/details-card.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { connect } from 'react-redux';
+
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '../../dashboard/dashboard-card';
+import { DetailsBody, DetailItem } from '../../dashboard/details-card';
+import { withDashboardResources, DashboardItemProps } from '../with-dashboard-resources';
+import { InfrastructureModel, ClusterVersionModel } from '../../../models';
+import { referenceForModel, K8sResourceKind, getOpenShiftVersion, getK8sGitVersion, getClusterName, ClusterVersionKind } from '../../../module/k8s';
+import { FLAGS } from '../../../const';
+import { featureReducerName, flagPending } from '../../../reducers/features';
+import { RootState } from '../../../redux';
+import { FirehoseResource } from '../../utils';
+
+const getInfrastructurePlatform = (infrastructure: K8sResourceKind): string => _.get(infrastructure, 'status.platform');
+
+const clusterVersionResource: FirehoseResource = {
+  kind: referenceForModel(ClusterVersionModel),
+  namespaced: false,
+  name: 'version',
+  isList: false,
+  prop: 'cv',
+};
+
+const infrastructureResource: FirehoseResource = {
+  kind: referenceForModel(InfrastructureModel),
+  namespaced: false,
+  name: 'cluster',
+  isList: false,
+  prop: 'infrastructure',
+};
+
+const mapStateToProps = (state: RootState) => ({
+  openShiftFlag: state[featureReducerName].get(FLAGS.OPENSHIFT),
+});
+
+export const DetailsCard_: React.FC<DetailsCardProps> = ({
+  watchURL,
+  stopWatchURL,
+  watchK8sResource,
+  stopWatchK8sResource,
+  resources,
+  urlResults,
+  openShiftFlag,
+}) => {
+  React.useEffect(() => {
+    if (!flagPending(openShiftFlag)) {
+      if (openShiftFlag) {
+        watchK8sResource(clusterVersionResource);
+        watchK8sResource(infrastructureResource);
+        return () => {
+          stopWatchK8sResource(clusterVersionResource);
+          stopWatchK8sResource(infrastructureResource);
+        };
+      }
+      watchURL('version');
+      return () => {
+        stopWatchURL('version');
+      };
+    }
+  }, [openShiftFlag, watchK8sResource, stopWatchK8sResource, watchURL, stopWatchURL]);
+
+  const clusterVersion = _.get(resources, 'cv');
```

Comment:
This seems cleaner to me:

```suggestion
  const clusterVersion = _.get(resources, 'cv.data') as ClusterVersionKind;
  // ...
  const openshiftVersion = getOpenShiftVersion(clusterVersion);
```

Reply:
Typescript complains `Conversion of type 'FirehoseResult' to type 'ClusterVersionKind' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.` . It thinks that `cv.data` is `FirehoseResult` but in reality it is `K8sResourceKind | K8sResourceKind[]` :/ Splitting the expression in a way I do does not result in error. Seems like a bug in TS.

Reply:
That seems weird. Is this a bug in the lodash typings? How does TypeScript know the type returned by `_.get`?

Reply:
I'd say that is an issue of `@types/lodash-es` but Im not very familiar with it. I tried updating to latest version (4.17.3, console is using 4.17.0) but it didnt help :/